### PR TITLE
Fixes for the cython port

### PR DESCRIPTION
**Pull request by jvkersch**
Third time's the charm: this is the old https://github.com/enthought/traits/pull/296, but against #294.

### CODE REVIEW

**Comment by mdickinson** [comment]
@jvkersch: That's better. Thank you!

File: traits/ctraits.pyx [hunk 1]
```diff
@@ -53,6 +54,8 @@ cdef object NullObject = object()
 # Python side.
 PY_NULL = object()
 
+is_callable = -1
```
Comment (dpinte) [inline traits/ctraits.pyx]: why not in uppercase? If I understand correctly, it's a global constant, no?

File: traits/ctraits.pyx [hunk 2]
```diff
@@ -404,9 +437,14 @@ cdef object validate_trait_complex(cTrait trait, CHasTraits obj, object name, ob
...
-    return validate_trait_tuple_check(trait.py_validate[1], obj, name, value)
+    cdef object result =  validate_trait_tuple_check(
+        trait.py_validate[1], obj, name, value
+    )
+    if result is not None:
+        return result
+    raise_trait_error(trait, obj, name, value)
 
-cdef object validate_trait_tuple_check(cTrait trait, CHasTraits obj, object name, object value):
+cdef object validate_trait_tuple_check(object trait, CHasTraits obj, object name, object value):
```
Comment (dpinte) [inline traits/ctraits.pyx]: why the change from `cTrait` to `object`?

**Comment by dpinte** [comment]
@jvkersch two minor comments, it looks great otherwise, thanks!

File: traits/ctraits.pyx (hunk 2 above)
Comment (jvkersch) [inline traits/ctraits.pyx]: `py_validate[1]` turns out to be a tuple sometimes. I don't know enough about traits to know how this works and what it means, but since the C implementation also has `object` I thought it would be appropriate to change to `object` (see https://github.com/enthought/traits/blob/master/traits/ctraits.c#L3378)

File: traits/ctraits.pyx (hunk 1 above)
Comment (jvkersch) [inline traits/ctraits.pyx]: I changed it to uppercase (it's indeed a global constant).

**Comment by jvkersch** [comment]
@dpinte Thanks for the swift review; I changed `is_callable` to uppercase and I looked at `validate_trait_tuple_check` again. I think this is ready for another round of review.

**Comment by dpinte** [comment]
@jvkersch 👍 merging!
